public/app/plugins/datasource/grafana-azure-monitor-datasource/components/NewMetricsQueryEditor/MetricsQueryEditor.tsx

```diff
@@ -38,83 +40,183 @@ const MetricsQueryEditor: React.FC<MetricsQueryEditorProps> = ({
   const metricsMetadata = useMetricMetadata(query, datasource, onChange);
   const metricNamespaces = useMetricNamespaces(query, datasource, onChange, setError);
   const metricNames = useMetricNames(query, datasource, onChange, setError);
-  return (
-    <div data-testid="azure-monitor-metrics-query-editor-with-resource-picker">
-      <InlineFieldRow>
-        <ResourceField
-          query={query}
-          datasource={datasource}
-          variableOptionGroup={variableOptionGroup}
-          onQueryChange={onChange}
-          setError={setError}
-          selectableEntryTypes={[ResourceRowType.Resource]}
-          setResource={setResource}
-          resourceUri={query.azureMonitor?.resourceUri}
-        />
-      </InlineFieldRow>
+  if (config.featureToggles.azureMonitorExperimentalUI) {
+    return (
+      <span data-testid="azure-monitor-metrics-query-editor-with-experimental-ui">
```

Comment:
nit: this testid will lose its meaning when we remove the feature flag so I would just use `azure-monitor-metrics-query-editor`